lesson_12/structs_ts/src/lesson12.ts

```diff
@@ -1,11 +1,26 @@
 import { ListNode } from './list_node.js';
 
 export class Lesson12 {
-  /**
-   * Provide the solution to LeetCode 3062 here:
-   * https://github.com/yang-su2000/Leetcode-algorithm-practice/tree/master/3062-winner-of-the-linked-list-game
-   */
+
   public gameResult(head: ListNode | null): string {
-    return '';
+    let [oddPoints, evenPoints] = [0,0];
+    let current = head;
+   
+   while (current != null && current.next != null) {
+      if (current.val > current.next.val) {
+        evenPoints++;
+      } else if (current.val < current.next.val) {
+        oddPoints++;
+      }
+      current = current.next ? current.next.next || null : null; 
```

Comment:
Reviewing these as documentation of struggles during this problem. Typescript complained a **_lot_** about whether something is undefined and how it was unsure. 

I referred to chatgpt and it showed me ways to ensure that my object is null so it can fix my issues. Especially right here I can recall the '?' is simply like the `if else` but in one line where it makes sure both options are null